website/project/model.py

```diff
@@ -2431,6 +2436,47 @@ def serialize(self):
             'is_registration': self.is_registration
         }
 
+    def _initiate_retraction(self, user, justification):
+        """Initiates the retraction process for a registration
+        :param user: User who initiated the retraction
+        :param justification: Justification, if given, for retraction
+        """
+
+        retraction = Retraction()
+        retraction.initiated_by = user
+        if justification:
+            retraction.justification = justification
+        retraction.initiation_date = datetime.datetime.utcnow()
+        retraction.state = 'pending'
+
+        # @todo(hrybacki) investigate why Node#admin_contributors isn't working
+        # Collect list of admins for registration
+        admins = [contrib for contrib in self.contributors if self.has_permission(contrib, 'admin')]
+
+        approval_state = {'num_of_approvals': 0}
+        # Create approve/disapprove keys
+        for admin in admins:
+            approval_state[admin._id] = {
+                'approval_token': security.random_string(30),
+                'disapproval_token': security.random_string(30),
+                'has_approved': False
+            }
+
+        retraction.approval_state = approval_state
+        return retraction
+
+    def retract_registration(self, user, justification=None):
+        """Retract public registration. Instantiate new Retraction object
+        and associate it with the respective registration.
+        """
+
+        if not self.is_public or not self.is_registration:
+            raise NodeStateError('Cannot retract private node or non-registration')
+
+        retraction = self._initiate_retraction(user, justification)
+        # Retraction record needs to be in the database for backrefs to work correctly
```

Comment:
Not quite accurate; the record needs to be saved in order to save the *forward* reference (i.e. the `retraction` foreign field).